app/controllers/posts_controller.rb

```diff
@@ -4,10 +4,21 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = Post.all.includes(:author).where(users: { id: params[:user_id] })
+
+    respond_to do |format|
+      format.html # index.html.erb
+      format.json { render json: @posts }
+    end
```

Comment:
:red_circle: [REQUIRED]  I've noticed that the response from the users/1/posts endpoint is not in JSON format. Instead, it's in HTML format.

![resul](https://github.com/microvese-projects/Blogit/assets/26736582/ae5eda7b-8b2a-4c00-807d-3a698cf0732b)

Kindly fix this issue

Reply:
@bienvenuushindi I would appreciate some clarification on the requirements. Are we supposed to have it in a way that the calls to the path /user/:id/posts can render both the HTML and the JSON or are we to render JSON only when that path is called?

Reply:
@Bennyjoez Thanks for your question. The requirements for the /user/:id/posts path are that it should only render JSON. This is because the path is being used by an API, and APIs should only return JSON data.

It's not a good practice to use the same controllers for both Rails apps and APIs. This is because the business logic for an API is often different from the business logic for a Rails app. By using separate controllers, you can keep the business logic for each type of application separate.

Here's how I would recommend you structure your controllers:

```
controllers
  |--api
  |  |-posts_controller.rb
  |  |-others_controller.rb
  |--app
  |  |-posts_controller.rb
  |  |-others_controller.rb
```

The api/posts_controller.rb and api/others_controller.rb controllers would be responsible for handling API requests for posts and others, respectively. The app/posts_controller.rb and app/others_controller.rb controllers would be responsible for handling requests for posts and users from the Rails app.

Use namespace and resources to define the routes for the API

```
# config/routes.rb

namespace :api do
   resources :users, only: [] do
           resources :posts, only: [:index, :show, :create] do
                resources :comments, only: [:index, :create]
           end 
  end 
end
```


I hope this helps! Let me know if you have any other questions.

Reply:
@BennyjoezThen you can create a separate parent controller for the API controllers, after that kindly make sure that all of the API controllers share the same behavior. This includes skipping the verify_authenticity_token and authenticate_user! filters, which are not necessary for API requests.

Here's the code for the ApiController parent controller:

```
class ApiController < ApplicationController
  skip_before_action :verify_authenticity_token
  skip_before_action :authenticate_user!
end

```

Then, the API controllers can inherit from the ApiController class like this:

```
class Api::PostsController < ApiController
  # ...
end

class Api::UsersController < ApiController
  # ...
end
```
I hope this helps! Let me know if you have any other questions.